gef.py

```diff
@@ -10424,24 +10466,27 @@ def __parse_gdb_info_sections(self) -> Generator[Section, None, None]:
                 off = int(parts[3][:-1], 16)
                 path = parts[4]
                 perm = Permission.from_info_sections(parts[5:])
-                yield Section(
-                    page_start=addr_start,
-                    page_end=addr_end,
-                    offset=off,
-                    permission=perm,
-                    inode="",
-                    path=path
-                )
+                yield Section(page_start=addr_start,
+                              page_end=addr_end,
+                              offset=off,
+                              permission=perm,
+                              path=path)
 
             except IndexError:
                 continue
             except ValueError:
                 continue
         return
 
-    def __parse_info_mem(self) -> Generator[Section, None, None]:
+    @staticmethod
+    def parse_monitor_info_mem() -> Generator[Section, None, None]:
         """Get the memory mapping from GDB's command `monitor info mem`"""
-        for line in StringIO(gdb.execute("monitor info mem", to_string=True)):
+        try:
+            stream = StringIO(gdb.execute("monitor info mem", to_string=True))
+        except Exception:
```

Comment:
this try/except is probably not necessary, let the exception propagate and the calling function handle it (or not)